### PR TITLE
[Common bugs] Fixed work shopping cart work with deleted course (CB-49)

### PR DESCRIPTION
**Pull request by Marx86**
**Description:**
Fix 404 error in the shopping cart, when a course was deleted with manage.py command.
Write tests for this issue.

**Youtrack:**
https://youtrack.raccoongang.com/issue/CB-49

### CODE REVIEW

**Comment by AndreyLykhoman** [comment]
start all tests

File: lms/djangoapps/shoppingcart/tests/test_views.py [hunk 1]
```diff
@@ -1488,9 +1488,34 @@ def test_shopping_cart_navigation_link_in_microsite_courseware_page(self):
             self.assertEqual(resp.status_code, 200)
             self.assertNotIn('<a class="shopping-cart"', resp.content)
 
+@attr(shard=3)
```
Comment (idegtiarov) [inline lms/djangoapps/shoppingcart/tests/test_views.py]: What does this attribute mean?

File: lms/djangoapps/shoppingcart/tests/test_views.py [hunk 2]
```diff
@@ -1488,9 +1488,34 @@ def test_shopping_cart_navigation_link_in_microsite_courseware_page(self):
...
+@patch.dict('django.conf.settings.FEATURES', {'ENABLE_PAID_COURSE_REGISTRATION': True})
+@ddt.ddt
+class ShoppingCartViewDeletedCourseTests(SharedModuleStoreTestCase, XssTestMixin):
+    """
+    Test shopping cart view deleted course under various states
```
Comment (idegtiarov) [inline lms/djangoapps/shoppingcart/tests/test_views.py]: The description is not clear enough could you please rephrase it to make more meaningful.

File: lms/djangoapps/shoppingcart/tests/test_views.py [hunk 3]
```diff
@@ -1488,9 +1488,34 @@ def test_shopping_cart_navigation_link_in_microsite_courseware_page(self):
...
+    """
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (idegtiarov) [inline lms/djangoapps/shoppingcart/tests/test_views.py]: Why do you need setUpClass method if there is only one test so far?

File: lms/djangoapps/shoppingcart/tests/test_views.py [hunk 4]
```diff
@@ -1487,6 +1488,48 @@ def test_shopping_cart_navigation_link_in_microsite_courseware_page(self):
...
+        self.assertEqual(resp.status_code, 200)
+        self.assertTrue(PaidCourseRegistration.contained_in_order(self.cart, self.course_key))
+
+        module_store = modulestore()
+        with module_store.bulk_operations(self.course_key):
+            module_store.delete_course(self.course_key, self.user)
+
+        resp = self.client.get(reverse('shoppingcart.views.show_cart', args=[]))
+        self.assertEqual(resp.status_code, 200)
+        ((template, context), _tmp) = render_mock.call_args
```
Comment (idegtiarov) [inline lms/djangoapps/shoppingcart/tests/test_views.py]: The outside parentheses are not required, remove them, please.

File: lms/djangoapps/shoppingcart/tests/test_views.py (hunk 1 above)
Comment (AndreyLykhoman) [reply lms/djangoapps/shoppingcart/tests/test_views.py]: `SHARD` is a number indicating which subset of the tests to build.
it was in ShoppingCartViewDeletedCourseTests class  because  ShoppingCartViewsTests class has same shard. And this installation provided testing of one entity in one section. 

Now I relocate test method  from ShoppingCartViewDeletedCourseTests to ShoppingCartViewsTests.

**Comment by hetmantsev** [comment]
start javascript tests
